### PR TITLE
Move prometheus, github, slack code from test-infra to pkg

### PR DESCRIPTION
**Pull request by Fredy-Z**
Move prometheus, github, slack related test code from `test-infra` to `pkg`.

/cc @srinivashegde86

### CODE REVIEW

File: test/prometheus/prometheus.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return v1.NewAPI(client), nil
+}
+
+// AllowPrometheusSync sleeps for sometime to allow prometheus time to scrape the metrics.
+func AllowPrometheusSync(logf logging.FormatLogger) {
+	logf("Sleeping to allow prometheus to record metrics...")
+	time.Sleep(30 * time.Second)
```
Comment (srinivashegde86) [inline test/prometheus/prometheus.go]: maybe this should be a param?

**Comment by Fredy-Z** [comment]
/cc @mattmoor  for approval.
Thanks

**Comment by Fredy-Z** [comment]
/cc @mattmoor 
for approval.
Thanks.

**Pull request merged**